command/agent/agent.go

```diff
@@ -314,9 +314,18 @@ func (a *Agent) clientConfig() (*clientconfig.Config, error) {
 	if conf == nil {
 		conf = clientconfig.DefaultConfig()
 	}
+
+	// If we are running a server, append both its bind and advertise address so
+	// we are able to at least talk to the local server even if that isn't
+	// configured explicitly. This handles both running server and client on one
+	// host and -dev mode.
+	conf.Servers = a.config.Client.Servers
 	if a.server != nil {
-		conf.RPCHandler = a.server
+		conf.Servers = append(conf.Servers,
+			a.config.Addresses.RPC,
```

Comment:
We don't check whether Addresses and AdvertiseAddrs could ever be nil- are these fields always set elsewhere in the code, or would it be valuable to have a check for nil beforehand?

Reply:
Shouldn't ever be nil as it is materialized well before we get here. See normalizeAddrs in command/command.go. Added safety guards